src/Storages/StorageMerge.cpp

```diff
@@ -216,6 +216,10 @@ Pipe StorageMerge::read(
     /// What will be result structure depending on query processed stage in source tables?
     Block header = getHeaderForProcessingStage(*this, column_names, metadata_snapshot, query_info, context, processed_stage);
 
+    /// There must be at least one column in the block so that it knows the number of rows.
+    if (!header)
+        header.insert({ColumnConst::create(ColumnUInt8::create(1, 0), 1), std::make_shared<DataTypeUInt8>(), "_dummy"});
```

Comment:
It doesn't look quite a good fix, and it also seems to cause the test `00022_merge_prewhere` to fail.

Reply:
+1.

However, we already rely on this fix in other place https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/evaluateConstantExpression.cpp#L47